x-pack/libbeat/reader/parquet/parquet.go

```diff
@@ -9,85 +9,110 @@ import (
 	}
+	log.Debugw("read data from stream reader", "size", len(data), "duration", fmt.Sprintf("%v", time.Since(start)))
 
 	// defines a memory allocator for allocating memory for Arrow objects
 	pool := memory.NewCheckedAllocator(&memory.GoAllocator{})
-
+	// constructs a parquet file reader object from the byte slice data
 	pf, err := file.NewParquetReader(bytes.NewReader(data), file.WithReadProps(parquet.NewReaderProperties(pool)))
 	if err != nil {
 		return nil, fmt.Errorf("failed to create parquet reader: %w", err)
 	}
+	log.Debugw("created parquet reader", "duration", fmt.Sprintf("%v", time.Since(start)))
 
 	// constructs a reader for converting to Arrow objects from an existing parquet file reader object
 	reader, err := pqarrow.NewFileReader(pf, pqarrow.ArrowReadProperties{
 		Parallel:  cfg.ProcessParallel,
-		BatchSize: int64(batchSize),
+		BatchSize: int64(cfg.BatchSize),
 	}, pool)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create pqarrow parquet reader: %w", err)
 	}
+	log.Debugw("created pqarrow parquet reader", "duration", fmt.Sprintf("%v", time.Since(start)))
 
 	// constructs a record reader that is capable of reding entire sets of arrow records
 	rr, err := reader.GetRecordReader(context.Background(), nil, nil)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create parquet record reader: %w", err)
 	}
+	log.Debugw("created parquet record reader", "duration", fmt.Sprintf("%v", time.Since(start)))
 
+	log.Debugw("initialization process completed", "duration", fmt.Sprintf("%v", time.Since(start)))
 	return &BufferedReader{
 		cfg:          cfg,
 		recordReader: rr,
 		fileReader:   pf,
+		log:          log,
 	}, nil
 }
 
+// initLogger initializes a logger for the parquet reader.
+func initLogger() *logp.Logger {
+	return logp.L().Named("reader.parquet")
+}
+
 // Next advances the pointer to point to the next record and returns true if the next record exists.
 // It will return false if there are no more records to read.
 func (sr *BufferedReader) Next() bool {
-	return sr.recordReader.Next()
+	next := sr.recordReader.Next()
+	if !next {
+		sr.log.Debugw("no more records to read", "next", next)
+	}
+	return next
 }
 
 // Record reads the current record from the parquet file and returns it as a JSON marshaled byte slice.
 // If no more records are available, the []byte slice will be nil and io.EOF will be returned as an error.
 // A JSON marshal error will be returned if the record cannot be marshalled.
 func (sr *BufferedReader) Record() ([]byte, error) {
+	start := time.Now()
 	rec := sr.recordReader.Record()
 	if rec == nil {
+		sr.log.Debugw("reached the end of the record reader", "record_reader", rec)
 		return nil, io.EOF
 	}
 	defer rec.Release()
 	val, err := rec.MarshalJSON()
 	if err != nil {
 		return nil, fmt.Errorf("failed to marshal JSON for parquet value: %w", err)
 	}
+	sr.log.Debugw("records successfully read", "batch_size", sr.cfg.BatchSize, "duration", fmt.Sprintf("%v", time.Since(start)))
```

Comment:
Not sure that we need the time since here. Note that having a `fmt.Sprint*` in a debug log is a source of load; we will always be doing the allocations for the formatting and so on, just to discard the work if DEBUG is not enabled.